lib/runner.ex

```diff
@@ -58,4 +53,24 @@ defmodule ElixirBench.Runner do
     data = Map.merge(data, job.context)
     {:ok, _} = Api.submit_results(client, job, data)
   end
+
+  defp build_job_struct(job_data) do
+    %{
+      "id" => id,
+      "repo_slug" => repo_slug,
+      "branch_name" => branch,
+      "commit_sha" => commit,
+      "config" => config
+    } = job_data
+
+    config = Config.from_string_map(config)
+
+    %Job{
+      id: to_string(id),
+      repo_slug: repo_slug,
+      branch: branch,
+      commit: commit,
+      config: config
+    }
```

Comment:
it's not immedeatly obvious what the benefit of building the job struct here is. Not sure it should be or how to make it better. Just an observation. I think we do it so that we have a job struct to work with rather than the data map. Then it's also weird though why some of the fields are called slightly different - we're in control of all those field names right? Or am I missing something? Then we should probably call them all the same, then creating a job struct from it is also much easier.